internal/k8s/builder/gateway.go

```diff
@@ -319,7 +330,17 @@ func (b *GatewayDeploymentBuilder) volumes() ([]corev1.Volume, []corev1.VolumeMo
 		volumes = append(volumes, corev1.Volume{
 			Name: "ca",
 			VolumeSource: corev1.VolumeSource{
-				EmptyDir: &corev1.EmptyDirVolumeSource{},
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: b.gateway.Name,
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "consul-ca-cert",
+							Path: consulCAFilename,
+						},
+					},
+					DefaultMode: nil,
+					Optional:    pointer.Bool(false),
+				},
```

Comment:
Instead of creating an empty volume and then having our shell script insert the CA cert, we now source the volume's contents from the `Secret` containing the CA cert, which is created further down in `internal/k8s/reconciler/deployer.go`